lib/test_prof/cops/rspec/aggregate_examples/its.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      class AggregateExamples < Cop
+        # @example `its`
+        #
+        #   # Supports regular `its` call with an attribute/method name,
+        #   # or a chain of methods expressed as a string with dots.
+        #
+        #   its(:one) { is_expected.to be(true) }
+        #   its('two') { is_expected.to be(false) }
+        #   its('phone_numbers.size') { is_expected.to be 2 }
+        #
+        # @example `its` with single-element array argument
+        #
+        #   # Also supports single-element array argument.
+        #
+        #   its(['headers']) { is_expected.to include(encoding: 'text') }
+        #
+        # @example `its` with multi-element array argument is ambiguous
+        #
+        #   # Does not support `its` with multi-element array argument due to
+        #   # an ambiguity. Transformation depends on the type of the subject:
+        #   # - a Hash: `hash[element1][element2]...`
+        #   # - and arbitrary type: `hash[element1, element2, ...]`
+        #   # It is impossible to infer the type to propose a proper correction.
+        #
+        #   its(['ambiguous', 'elements']) { ... }
+        #
+        # @example `its` with metadata
+        #
+        #   its('header', html: true) { is_expected.to include(text: 'hello') }
+        #
+        module Its
+          extend RuboCop::NodePattern::Macros
+
+          private
+
+          # It's impossible to aggregate `its` body as is, it needs to be
+          # converted to `expect(subject.something).to ...`
+          def new_body(node)
```

Comment:
It's not too obvious (you can see `super` being used, and a mess of `include`/`prepend` in the cop base body) that some of the methods are overridden.
It makes sense to cross-reference methods that override each other.

Reply:
What about chaining the naming of the module? For example, `Aggregation` -> `BaseAggregation` (which indicates that this module should be included first). Similarly, `Its` -> `ItsAggregation` (hints that it extends `BaseAggregation`).